llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/SeedCollector.h

```diff
@@ -284,6 +284,36 @@ class SeedContainer {
 #endif // NDEBUG
 };
 
+class SeedCollector {
+  SeedContainer StoreSeeds;
+  SeedContainer LoadSeeds;
+  BasicBlock *BB;
+  Context &Ctx;
+
+  /// \Returns the number of SeedBundle groups for all seed types.
+  /// This is to be used for limiting compilation time.
+  unsigned totalNumSeedGroups() const {
+    return StoreSeeds.size() + LoadSeeds.size();
+  }
+
+public:
+  SeedCollector(BasicBlock *SBBB, ScalarEvolution &SE);
+  ~SeedCollector();
+
+  BasicBlock *getBasicBlock() { return BB; }
```

Comment:
I suspect it will be useful for dumping, and possibly to manage the remove instruction callback somewhat more intelligently than registering and unregistering every basic. But block, but for now it has no uses.

Reply:
Shouldn't we remove it until we have a use for it then?